### PR TITLE
[resolver] use webpack-resolver for core logic

### PR DESCRIPTION
**Pull request by spalger**
This PR removes the custom resolver logic and instead wraps the `eslint-import-resolver-webpack` module, feeding it a custom configuration so that the resolver behaves like Kibana's internal version of webpack.

This also adds three settings:
- `rootPackageName`: the name of the package that should be considered the "root" package, useful for packages like Kibana which contain multiple packages so that a file like `src/core_plugins/console/index.js` resolves as a child of kibana rather than the console plugin package.
- `pluginPaths`: an array of paths to plugin folders
- `pluginDirs`: an array of paths (relative to the project root) that contain multiple plugin folders

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,214 +1,106 @@
...
+const webpack = require('webpack');
+const DirectoryNameAsMain = require('@elastic/webpack-directory-name-as-main');
 const debug = require('debug')('eslint-import-resolver-kibana');
 
 const defaults = {
   kibanaPath: '../kibana',
+  pluginPaths: [],
+  pluginDirs: [
+    '../kibana/plugins',
+    '../kibana/core_plugins',
```
Comment (w33ble) [inline index.js]: This path is wrong, it's `kibana/src/core_plugins`

**Review by w33ble — CHANGES_REQUESTED** [review]

File: index.js [hunk 2]
```diff
@@ -1,213 +1,130 @@
...
-  const globOptions = {
-    cwd: path.resolve(checkPath),
-  };
+function getProjectRoot(file, config) {
+  const { rootPackageName } = config;
 
-  const matches = glob.sync(globPattern, globOptions);
-  debug(`checking in ${checkPath}, matched ${matches.length}`);
+  const getRootPackageDir = (dir) => {
+    const pkgFile = pkgUp.sync(dir);
```
Comment (spalger) [inline index.js]: Do we really need `pkgUp` for this? We're basically already doing what it is doing in this function.

Comment (spalger) [reply index.js]: Start at the path and recursively look for package.json files that match the `rootPackageName` or just exist if there isn't a `rootPackageName`.

Comment (w33ble) [reply index.js]: I suppose you're right, we kinda don't...

File: index.js [hunk 3]
```diff
@@ -1,213 +1,130 @@
...
   kibanaPath: '../kibana',
+  pluginDirs: [],
+  pluginPaths: [],
 };
 
 /*
  * Resolves the path to Kibana, either from default setting or config
  */
-function getKibanaPath(config, file, rootPath) {
+function getKibanaPath(config, rootPath) {
```
Comment (spalger) [inline index.js]: Can we rename `rootPath` to `projectRoot`?

**Review by w33ble — APPROVED** [review]

**Pull request merged**